ast-nodes/type-declaration.dart

```diff
@@ -41,6 +42,11 @@ class TypeDeclaration extends Declaration {
   }
 
   void checkSemantics() {
-    // TODO: implement
+    value.checkSemantics();
+
+    var declaration = this.scopeMark.resolve(this.name);
+    if (declaration is! null) {
```

Comment:
I don't think this will work. There's a utility function specifically for this purpose, `ensureNoOther`. See `routine-declaration`